tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_api.py

```diff
@@ -85,6 +85,47 @@ def test_save_aliases_after_logging_artifact(user, wandb_init):
     assert "hello" in aliases
 
 
+def test_save_tags_after_logging_artifact(tmp_path, user, wandb_init, api):
+    project = "test"
+    artifact_name = "test-artifact"
+    artifact_type = "test-type"
+    artifact_fullname = f"{user}/{project}/{artifact_name}:v0"
+
+    artifact_filepath = tmp_path / "boom.txt"
+    artifact_filepath.write_text("testing")
+
+    orig_tags = ["orig-tag", "other-tag"]  # Initial tags on the logged artifact
+    to_delete = ["other-tag"]  # Tags to delete later on
+    to_add = ["added-tag"]  # Tags to add later on
+
+    with wandb_init(entity=user, project=project) as run:
+        artifact = wandb.Artifact(name=artifact_name, type=artifact_type)
+        artifact.add_file(str(artifact_filepath), "test-name")
+
+        # Assign tags when logging
+        run.log_artifact(artifact, tags=orig_tags)
+        artifact.wait()
+
```

Comment:
Maybe we should add an assert here that the tags were successfully attached when calling log_artifact()

Reply:
Definitely -- thanks for the catch!